core/sql/optimizer/ControlDB.cpp

```diff
@@ -108,7 +108,12 @@ ControlDB::~ControlDB()
 
 void ControlDB::setRequiredShape(ControlQueryShape *shape)
 {
-  delete requiredShape_;
+  if (requiredShape_)
+    {
+      delete requiredShape_->getShape();
+      delete requiredShape_;
```

Comment:
If the NADELETE macro can't be used, then it is fine to stick with delete. However, NABasicObject also has the default constructor and hence it is possible to create objects without heap. I guess overloaded delete operator would take care of this too.

I preferred NADELETE macro because it is easy to read that it is being deleted from the heap.